### ISSUE TITLE
2.4 Wish List

### ISSUE THREAD

**Issue by rande** [sonata-project/SonataAdminBundle#2406]
Validated elements:

- [x] Mozaic view list
- [ ] Filter on top (something like redmine)

Please comment to add your ideas.

**Comment by meniku** [comment]
+1 for #1971

**Comment by zhil** [comment]
@na-ji  :+1:  nice feature!

**Comment by amine26tz** [comment]
@na-ji look at #2711

**Comment by na-ji** [comment]
@amine26tz Quite nice what you did there, congrats ! :+1: Waiting for your PR to be merged.

**Comment by krewetka** [comment]
+1 for #1971 ...preferably in 2.3 as well :)

**Comment by YuanluDWD** [comment]
+1 for  #1382, when could this be accomplished?

**Comment by krewetka** [comment]
@YuanluDWD - as far as I know in master version filters are on a top. I had to check it once and have to move back to 2.3 just because of changed location of filters. Did you try the newest version?

**Comment by YuanluDWD** [comment]
@krewetka Thanks for your answer. I will discuss it with our engineers.

**Comment by krewetka** [comment]
@YuanluDWD - if it's not your personal project I guess the best option would be wait until stable 2.4 tag as current master version is as always work in progress. 

Still you can use it to see how filters will look in new version.

**Comment by rande** [comment]
@Storke the getSubject() should now return the correct value: https://github.com/sonata-project/SonataAdminBundle/commit/1a5ba565a75c6c43867c52210cb78ec29a2b6748

**Comment by webdevilopers** [comment]
@Storke @rande With current `dev-master` I can call `getSubject()`. But I don't get an object returned when calling `getSubject()->getParent()` neither does `isChild()` return true though my admin class is a child.

```php
class ContractAdmin extends Admin
{
    protected $parentAssociationMapping = 'branch';

    protected function configureFormFields(FormMapper $formMapper)
    {
        $formMapper
                ->add('supplierContacts', 'sonata_type_collection', array(
                       'by_reference' => false // !important
                    ), array(
                        'edit' => 'inline',
                        'inline' => 'table'
                    ));
    }
}
```

```php
class SupplierContactAdmin extends Admin
{
    protected $parentAssociationMapping = 'contract';

    protected function configureFormFields(FormMapper $formMapper)
        if (!$this->isChild()) {
         } else {
            //$contract = $this->getParent()->getSubject();
            $contract = $thisgetSubject()->getParent();
            echo $contract->getNumber();
        }
    }
}
```

Is this an issue with nested admins with depth > 1?

Possible related to:
- https://github.com/sonata-project/SonataAdminBundle/issues/2077
- https://github.com/sonata-project/SonataAdminBundle/issues/2356
- https://github.com/sonata-project/SonataAdminBundle/issues/2847

**Comment by Soullivaneuh** [comment]
@rande How about Symfony version support?

SonataAdmin 2.4 stable is not out yet and symfony 2.7 will be out shortly (May 2017) with 2.3 **end of support**.

I think it's not unreasonable to drop symfony 2.3+ support and require 2.7+ version.

This will permit to fix get free of a lot deprecated calls.

Regards.

**Comment by GromNaN** [comment]
:+1:

**Comment by Soullivaneuh** [comment]
Forget my idea about 2.3, this version is still supported until may 2017.

**Comment by Soullivaneuh** [comment]
Can be closed as 3.0 is finally out.

Please open separated issues instead.
